tests/integration/test_github_workflows.py

```
@@ -1888,4 +1923,862 @@
+        )
+
+class TestPRAgentWorkflowAdvanced:
+    """Advanced tests for PR Agent workflow configuration and security."""
+    
+    def test_no_duplicate_keys_in_workflow(self, workflow_files):
+        """Ensure no duplicate YAML keys exist (the bug that was fixed)."""
+        pr_agent_workflow = workflow_files.get('.github/workflows/pr-agent.yml')
+        assert pr_agent_workflow is not None, "pr-agent.yml not found"
+        
+        # Parse and check for duplicate keys at each level
+        def check_duplicates(node, path=""):
```

Ruff (0.14.6)

Missing return type annotation for private function `check_duplicates`

Add return type annotation: `None`

(ANN202)